src/bpftrace.cpp

```diff
@@ -643,6 +643,11 @@ std::vector<std::unique_ptr<IPrintable>> BPFtrace::get_arg_values(const std::vec
             std::string(p, strnlen(p, arg.type.size))));
         break;
       }
+      case Type::buffer:
+        arg_values.push_back(std::make_unique<PrintableString>(resolve_buf(
+            reinterpret_cast<char *>(arg_data + arg.offset + sizeof(uint8_t)),
+            reinterpret_cast<uint8_t>(*(arg_data + arg.offset)))));
```

Comment:
I think it would be nicer and less error-prone if we reused `AsyncEvent::Buf` here:
```suggestion
            reinterpret_cast<AsyncEvent::Buf *>(arg_data + arg.offset)->content,
            reinterpret_cast<AsyncEvent::Buf *>(arg_data + arg.offset)->length)));
```

Note this suggestion only works if we go with the FLA as suggested last commit

Reply:
Yes, good call. This feels much better.